README.md

```diff
@@ -1,16 +1,65 @@
+[![Hits-of-Code](https://hitsofcode.com/github/artipie/cargo-adapter)](https://hitsofcode.com/view/github/artipie/cargo-adapter)
+[![Maven Central](https://img.shields.io/maven-central/v/com.artipie/cargo-adapter.svg)](https://maven-badges.herokuapp.com/maven-central/com.artipie/cargo-adapter)
+[![PDD status](http://www.0pdd.com/svg?name=artipie/cargo-adapter)](http://www.0pdd.com/p?name=artipie/cargo-adapter)
 
-Cargo manages project dependencies, compile sources and distribute arttifdacts to repositories.
+This Java library turns your binary storage (files, S3 objects, anything) into Cargo repository - 
+Rust Package Registry. You may add it to your binary storage and it will become a fully-functionable 
+repository, which [cargo](https://doc.rust-lang.org/book/ch01-03-hello-cargo.html) will perfectly understand.
 
-Cargo [guide](https://doc.rust-lang.org/cargo/guide/index.html).
-Cargo [reference](https://doc.rust-lang.org/cargo/reference/index.html).
+Some valuable references and need-to-know:
+- Cargo is a [Rust](https://www.rust-lang.org/) package manager. Cargo manages project dependencies, 
+compile sources and distribute arttifdacts to repositories. 
+- Rust [package](https://doc.rust-lang.org/cargo/appendix/glossary.html#package)
+- Cargo [guide](https://doc.rust-lang.org/cargo/guide/index.html).
+- Cargo [reference](https://doc.rust-lang.org/cargo/reference/index.html).
+- Default Cargo repository is https://crates.io, Sources of crates.io: https://github.com/rust-lang/crates.io
+- Start Crates with docker-compose: https://github.com/rust-lang/crates.io/blob/master/docs/CONTRIBUTING.md#running-cratesio-with-docker
 
-Default Cargo repository is https://crates.io
+# Repository index format
+Repository indexes are stored in the git repository, which allows cargo efficiently fetch incremental updates. 
+In the repository root `config.json` should be located, this file contains the info for accessing the 
+registry:
+```json
+{
+    "dl": "https://crates.io/api/v1/crates",
+    "api": "https://crates.io"
+}
+```
+where `dl` is the url for downloading crates and `api` is the base URL for the web API, 
+`api` is optional, but without it commands such `cargo publish` will not work.
 
-Sources of crates.io: https://github.com/rust-lang/crates.io
+Download url should send `.crate` file for the requested package, package name and version are appended 
+to the base url along with `downlowd` word: `{package_name}/{package_version}/download`. `dl` value
+also may have markers in the url, find more information [here](https://doc.rust-lang.org/cargo/reference/registries.html#index-format).
 
-Start Crates with docker-compose: https://github.com/rust-lang/crates.io/blob/master/docs/CONTRIBUTING.md#running-cratesio-with-docker
+The rest of the repository index contains one file for each package, where each version metadata are
```

Comment:
typo for `each version metadata are present` --> `each version metadata is presented`